.github/workflows/_Test-OCI-Factory.yaml

```
@@ -15,7 +15,58 @@ on:
       - "!src/workflow-engine/**"
       - "!src/cli-client/**"
 
+env:
+  # local path to clone the oci-factory to
+  OCI_FACTORY_DIR: oci-factory/
+
+  # path of pytest junit output
+  PYTEST_RESULT_PATH: pytest_results.xml
+
+
 jobs:
+
+  pytest:
+    # Trigger python unit tests across the repository
+    name: pytest
+    runs-on: ubuntu-22.04
+    steps:
+
+      # Job Setup
+      - uses: actions/checkout@v4
+        with:
+          path: ${{ env.OCI_FACTORY_DIR }}
+          fetch-depth: 1
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+      
+      # Note: Add additional dependency installation lines as required below 
+      # test-oci-factory/pytest requirements
+      - run: pip install -r ${{ env.OCI_FACTORY_DIR }}/src/test-oci-factory/pytest/requirements.txt
```

```suggestion
      - run: pip install -r src/test-oci-factory/pytest/requirements.txt
```

(same for other occurrences below)